### ISSUE TITLE
how to generate change map 

### ISSUE THREAD

**Issue by YangTze7** [lehaifeng/DASNet#9]
Hi,I'm new to CD task,recently I've read your paper and test the code,as change map shown in your paper,it is binary image,but I can only get distance map according to your description of the method.Could you give some guide?Appreciate your help!
![image](https://user-images.githubusercontent.com/30680563/81055997-5d832500-8efc-11ea-808c-d0469de946da.png)

**Comment by eedavidwu** [comment]
> Hi,I'm new to CD task,recently I've read your paper and test the code,as change map shown in your paper,it is binary image,but I can only get distance map according to your description of the method.Could you give some guide?Appreciate your help!
> ![image](https://user-images.githubusercontent.com/30680563/81055997-5d832500-8efc-11ea-808c-d0469de946da.png)

Hi， the distance map is proportional to the distance map (or it could be achieved in the single_layer_similar_heatmap_visual)...
(BTW, have u trained and test this net successfully? It seems this net doesn't converge in loss with the pretrained net VGG or CDD pretrained model it given. Or any tricks needed?)

**Comment by YangTze7** [comment]
> > Hi,I'm new to CD task,recently I've read your paper and test the code,as change map shown in your paper,it is binary image,but I can only get distance map according to your description of the method.Could you give some guide?Appreciate your help!
> > ![image](https://user-images.githubusercontent.com/30680563/81055997-5d832500-8efc-11ea-808c-d0469de946da.png)
> 
> Hi， the distance map is proportional to the distance map (or it could be achieved in the single_layer_similar_heatmap_visual)...
> (BTW, have u trained and test this net successfully? It seems this net doesn't converge in loss with the pretrained net VGG or CDD pretrained model it given. Or any tricks needed?)

"the distance map is proportional to the distance map" Can u describe more about that？I do not get it from the paper.Also in my experiment ,the loss looks  strange,and what version of Pytorch you use?my environment is Pytorch 1.3 
![image](https://user-images.githubusercontent.com/30680563/81191884-1d9b6b00-8fec-11ea-9bec-5512e9f8429d.png)

**Comment by lehaifeng** [comment]
you can use single_layer_similar_heatmap_visual to generate change map

**Comment by eedavidwu** [comment]
> > > Hi,I'm new to CD task,recently I've read your paper and test the code,as change map shown in your paper,it is binary image,but I can only get distance map according to your description of the method.Could you give some guide?Appreciate your help!
> > > ![image](https://user-images.githubusercontent.com/30680563/81055997-5d832500-8efc-11ea-808c-d0469de946da.png)
> > 
> > 
> > Hi， the distance map is proportional to the distance map (or it could be achieved in the single_layer_similar_heatmap_visual)...
> > (BTW, have u trained and test this net successfully? It seems this net doesn't converge in loss with the pretrained net VGG or CDD pretrained model it given. Or any tricks needed?)
> 
> "the distance map is proportional to the distance map" Can u describe more about that？I do not get it from the paper.Also in my experiment ,the loss looks strange,and what version of Pytorch you use?my environment is Pytorch 1.3
> ![image](https://user-images.githubusercontent.com/30680563/81191884-1d9b6b00-8fec-11ea-9bec-5512e9f8429d.png)

So have u tested this code in right results?...
For that I check the torch version but loss seems not right.

**Comment by YangTze7** [comment]
@eedavidwu I have run this repo's code  with Pytorch 1.4, the loss  is strange.Is the version of exp your 
 environment Pytorch 0.3?

**Comment by eedavidwu** [comment]
> @eedavidwu I have run this repo's code with Pytorch 1.3, the loss is strange.Is the version of your experimental environment 0.3？
Yes i changed the version into 0.3, But my loss keeps around the 1.0...
So did u change your learning rate or sth else?
And is the F1 score or other printed number in the validate process right or not?

**Comment by TingquanGao** [comment]
> @eedavidwu I have run this repo's code with Pytorch 1.3, the loss is strange.Is the version of your experimental environment 0.3？

Hi,
Could you get the result of binary image?

**Comment by eedavidwu** [comment]
> > @eedavidwu I have run this repo's code with Pytorch 1.3, the loss is strange.Is the version of your experimental environment 0.3？
> 
> Hi,
> Could you get the result of binary image? I meet the same question.

The result of the net is the distance between two maps, and if you want to get the binary result, you need to do some process with the output.
Maybe you can add one layer or set one threshold.

**Comment by TingquanGao** [comment]
Thanks for your anwser.
I have done some processing on the distance map, for example, `MinMaxScaler([0, 255])`, `threshold()`. However, the result is unsatisfactory. 
Would you get the performance in the paper? And, what pytorch version are you doing?

**Comment by eedavidwu** [comment]
. I used the backbone and havn't computed the OA,PRE,RECALL..
I think the changemap can be generated butI cannot make sure the performance good enough.
The 'unsatisfactory' means: Not good enough or not acceptable?

**Comment by TingquanGao** [comment]
I also did not compute OA,PRE,RECALL..
I mean that the distance map does not seem to achieve the effect in the paper with model provided by author, like this:

<img width="356" alt="截屏2020-05-23 下午8 01 50" src="https://user-images.githubusercontent.com/12560511/82730157-46cc3300-9d30-11ea-96ca-ea26a4c8afe9.png">

**Comment by eedavidwu** [comment]
> I also did not compute OA,PRE,RECALL..
> I mean that the distance map does not seem to achieve the effect in the paper with model provided by author, like this:
> 
> <img alt="截屏2020-05-23 下午8 01 50" width="356" src="https://user-images.githubusercontent.com/12560511/82730157-46cc3300-9d30-11ea-96ca-ea26a4c8afe9.png">

Any pics of your result? I have checked the distance map, it has some small noise like sawtooth.

**Comment by TingquanGao** [comment]
I tested the CDD dataset using `BCD_model_best.pth` provided by author. Some results as follow:
![1](https://user-images.githubusercontent.com/12560511/82731384-ff967000-9d38-11ea-88e3-02098713651e.png)
![2](https://user-images.githubusercontent.com/12560511/82731399-2359b600-9d39-11ea-8847-ad888ae0e6ff.png)

**Comment by louxiantuo** [comment]
> PRE

Have you solved the problem of negative loss and non-convergence?
